Gom4ziz/Source/Data/Repository/FirebaseQuestionAnswerRepository.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  FirebaseQuestionAnswerRepository.swift
+//  Gom4ziz
+//
+//  Created by 이가은 on 2022/11/24.
+//
+
+import Foundation
+
+import FirebaseFirestore
+import RxSwift
+
+protocol QuestionAnswerRepository {
+    func fetchQuestionAnswerList(for userId: String, before artworkId: Int) -> Observable<[QuestionAnswer]>
+}
+
+struct FirebaseQuestionAnswerRepository {
+    static let shared = FirebaseQuestionAnswerRepository()
+    private let db: Firestore = Firestore.firestore()
+}
+
+// MARK: - QuestionAnswerRepository protocol extension
+extension FirebaseQuestionAnswerRepository: QuestionAnswerRepository {
+    func fetchQuestionAnswerList(for userId: String, before artworkId: Int) -> Observable<[QuestionAnswer]> {
+        var questionAnswerObservableList: [Observable<QuestionAnswer>] = []
+        for i in 1...artworkId {
+            questionAnswerObservableList.append(getQuestionAnswerRef(for: userId, before: i)
```

Comment:
[사소한 의견]

수고하셨습니다!

for 문으로 배열을 돌면서 옵저버블을 추가하기보단, 

artworkId 배열에 map 함수를 적용해서 옵저버블 배열을 만들면 훨씬 깔끔할 것 같아요!!
```suggestion
[1...artworkId].map { getObservable() }
```

Reply:
저도 사실 처음에는 이 코드로 짰었는데,
map 안에서 error가 발생되면, 성공한 애들도 다 같이 날라가 버리더라구요.. closure 관련 issue인 것 같은데.. 잘 모르겠어서 우선 위와 같이 구현했습니다.. closure 공부해서 다시 try 해보겠습니다! 제안 감사합니다 ☺️
error 예시) 
artworkId 값이 DB에 있는 Id 값보다 클 경우
```
        let questionAnswerObservableList = Array(1...artworkId).map { i in
            getQuestionAnswerRef(for: userId, before: i)
                .rx
                .decodable(as: QuestionAnswer.self)
        }
```